### PR TITLE
Feat/545/quick filters

### PR DESCRIPTION
**Pull request by oliversberzs**
# Issue

#545

## Description

* Added quick filters section to home page

### CODE REVIEW

File: packages/ide-extension/src/panel/guidedAnswersPanel.ts [hunk 1]
```diff
@@ -161,9 +161,8 @@ export class GuidedAnswersPanel {
         }
         if (this.startOptions) {
             await this.processStartOptions(this.startOptions);
-        } else {
-            await this.processEnvironmentFilters(this.ide);
         }
+        await this.loadQuickFilters(this.ide);
```
Comment (Klaus-Keller) [inline packages/ide-extension/src/panel/guidedAnswersPanel.ts:165]: If Guided Answers extension is started with arguments (`this.startOptions`) to load a certain tree and node path, I think we should show what was passed as argument and NOT overwrite it with quickfilters. That is why it previously was in the `else`-branch.

**Review by Klaus-Keller — CHANGES_REQUESTED** [review]
See my comments

File: packages/core/test/guided-answers-api.test.ts [hunk 2]
```diff
@@ -864,7 +864,7 @@ describe('Guided Answers Api: getNodePath()', () => {
         expect(result.length).toBe(2);
         expect(result[0].BODY).toBe('<p>Should not be modified at all</p>');
         expect(result[1].BODY).toBe('<p>Also should not be modified at all</p>');
-        expect(options.logger?.logString).toBeCalledWith(expect.stringContaining('Unexpected end of JSON input'));
+        expect(options.logger?.logString).toBeCalledWith(expect.stringContaining('Error when parsing argument'));
```
[the remark this answers is not in the document]
Comment (oliversberzs) [reply packages/core/test/guided-answers-api.test.ts:867]: I noticed that the error that the JSON parsing throws is has a different message on different node versions, so it is a lot safer and consistent to test not for the error message that node throws, but the specific one we are throwing
image

File: packages/ide-extension/src/panel/guidedAnswersPanel.ts (hunk 1 above)
Comment (Klaus-Keller) [reply packages/ide-extension/src/panel/guidedAnswersPanel.ts:165]: Great find, and also a good reminder that we need to extend the node versions for pipeline builds. Currently only 
```
node-version: [16.x, 18.x]
```
https://github.com/SAP/guided-answers-extension/blob/5af480cd96bf64e31bc3cc139906630ecc515bb4/.github/workflows/pipeline.yml#L14C20-L14C20

I'll create a separate PR for this.

File: packages/core/test/guided-answers-api.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Klaus-Keller) [reply packages/core/test/guided-answers-api.test.ts:867]: Great find, and also a good reminder that we need to extend the node versions for pipeline builds. Currently only 
```
node-version: [16.x, 18.x]
```
https://github.com/SAP/guided-answers-extension/blob/5af480cd96bf64e31bc3cc139906630ecc515bb4/.github/workflows/pipeline.yml#L14C20-L14C20

I'll create a separate PR for this.

File: packages/ide-extension/src/panel/guidedAnswersPanel.ts (hunk 1 above)
Comment (Klaus-Keller) [reply packages/ide-extension/src/panel/guidedAnswersPanel.ts:165]: Sorry, I missed the part that behavior of quick filters slightly changed, they are not loaded on start anymore. Saw that now in the ticket.

**Review by Klaus-Keller — APPROVED** [review]
- code changes look good
- test coverage is good
- tested locally, but not on SBAS

Approved from my side.

File: packages/webapp/src/webview/ui/components/QuickFilters/QuickFilters.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ * @returns - react elements for QuickFilters view
+ */
+export function QuickFilters(): ReactElement {
+    const filters = useSelector<AppState, GuidedAnswersQueryFilterOptions[]>((state) => state.quickFilters);
+    return (
+        <div>
+            <h3 style={{ display: 'flex', alignItems: 'center' }}>
+                <VscFilter />
+                <span style={{ margin: '0 5px' }}>Quick Filters</span>
```
Comment (unseen1980) [inline packages/webapp/src/webview/ui/components/QuickFilters/QuickFilters.tsx:22]: @oliversberzs Can you please in one of your next PRs update here the text with i18n values? Thanks
https://github.com/SAP/guided-answers-extension/blob/main/packages/webapp/src/webview/i18n/i18n.json

**Pull request merged**